### PR TITLE
add edit url to tables

### PR DESCRIPTION
**Pull request by julsbreakdown**
List of checkboxes have a new pen icon to allow users to edit an element (for instance, a
role). Clicking on this icon will first check if changes were made in the current form. If the
user switches to another form, there's no way back – only one form is open at a time

Requires:
Adapt templates : 
https://github.com/camptocamp/c2cgeoform/pull/153
Check before leaving page if form has changed:
https://github.com/camptocamp/c2cgeoform/pull/155

### CODE REVIEW

File: admin/c2cgeoportal_admin/views/themes.py [hunk 1]
```diff
@@ -35,7 +35,12 @@
             ]).alias('functionnality_labels'),
             'id',
             'label',
-            order_by='label'
+            order_by='label',
+            edit_url=lambda request, value: request.route_url(
+                'c2cgeoform_item',
+                table='restricted_roles',
```
Comment (arnaud-morvan) [inline admin/c2cgeoportal_admin/views/themes.py]: Seems it should be functionalities, not restricted_roles

File: admin/c2cgeoportal_admin/views/themes.py [hunk 2]
```diff
@@ -49,7 +54,12 @@
             Role,
             'id',
             'name',
-            order_by='name'
+            order_by='name',
+            edit_url=lambda request, value: request.route_url(
+                'c2cgeoform_item',
+                table='restricted_roles',
```
Comment (arnaud-morvan) [inline admin/c2cgeoportal_admin/views/themes.py]: Seems it should be roles, not restricted_roles

File: admin/acceptance_tests/__init__.py [hunk 3]
```diff
@@ -86,6 +86,14 @@ def _check_restrictionsareas(self, form, ras, item):
...
             } for ra in sorted(ras, key=lambda ra: ra.name)])
 
+    def get_edit_relationships(self):
+        return {
+            'layers_wmts': ['restrictionareas', 'interfaces'],
+            'layers_wms': ['restrictionareas', 'interfaces'],
+            'layers_v1': ['restrictionareas', 'interfaces'],
+            'roles': ['restrictionareas', 'functionalities']
+            }
+
```
Comment (arnaud-morvan) [inline admin/acceptance_tests/__init__.py]: could be moved in edit_url_test.py

**Comment by julsbreakdown** [comment]
@sbrunner 
any update on codacy?

**Comment by arnaud-morvan** [comment]
@sbrunner: this PR is blocked by codacy, we do not have rights to remove the issue in codacy, it seems that only you can remove it. Removing the required Codacy green status should solve the problem.

**Comment by sbrunner** [comment]
Fixed :-)

**Comment by julsbreakdown** [comment]
thanks

**Pull request merged**